controllers/configurationpolicy_controller.go

```diff
@@ -1688,22 +1663,34 @@ func (r *ConfigurationPolicyReconciler) handleSingleObj(
 ) {
 	objLog := log.WithValues("object", obj.name, "policy", obj.policy.Name, "index", obj.index)
 
+	result = objectTmplEvalResult{
+		objectNames: []string{obj.name},
+		namespace:   obj.namespace,
+		events:      []objectTmplEvalEvent{},
+	}
+
 	if !exists && obj.shouldExist {
 		// object is missing and will be created, so send noncompliant "does not exist" event regardless of the
 		// remediation action
-		noncompliantReason := generateSingleObjReason(obj.shouldExist, false, exists)
-		result = objectTmplEvalResult{
-			objectNames: []string{obj.name},
-			namespace:   obj.namespace,
-			events:      []objectTmplEvalEvent{{false, noncompliantReason, ""}},
-		}
+		result.events = append(result.events, objectTmplEvalEvent{false, reasonWantFoundDNE, ""})
 
 		// it is a musthave and it does not exist, so it must be created
 		if strings.EqualFold(string(remediation), string(policyv1.Enforce)) {
 			var uid string
 			completed, reason, msg, uid, err := r.enforceByCreatingOrDeleting(obj)
 
-			result.events = append(result.events, objectTmplEvalEvent{completed, reason, msg})
+			hasStatus := false
+			if tmplObj, err := unmarshalFromJSON(objectT.ObjectDefinition.Raw); err == nil {
+				_, hasStatus = tmplObj.Object["status"]
+			}
+
+			if completed && hasStatus {
+				msg += ", but has a status that might not be correct yet"
```

Comment:
How about this?
```suggestion
				msg += ". The status will be validated on the next evaluation."
```

Reply:
Instead, perhaps `msg += ". The status will be verified on the next evaluation."`

Reply:
Good suggestion! I've reworded it slightly again, and I wanted to keep the comma: if a policy has multiple objects, the messages will be combined with semicolons, so using a period in the middle of a message seemed like it could be weird.

I also adjusted the `reason` to be shorter now.